File: API/models/meal.ts
import { model, Schema } from 'mongoose';

export interface Meal {
  mealId: String;
  userId: String;
  title: String;
}

const mealSchema = new Schema<Meal>({
  mealId: { type: String, required: true },
  userId: { type: String, required: true },
  title: { type: String, required: false },
});

export const Meal = model<Meal>('Meal', mealSchema);

File: API/internal/models/meal.go
package model

import (
	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Meal struct {
	gorm.Model
	ID uuid.UUID `gorm:"type:uuid"`
	UserId string `json:"userId"`
	MealId string `json:"mealId"`
	Title string `json:"title"`
}